implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -219,6 +205,30 @@
 
         String value = configValue.getValue(); // Can return the empty String (which is not considered as null)
 
+        return convertValue(name, value, converter);
+    }
+
+    /**
+     * 
+     * This method handles converting values for both CDI injections and programatical calls.<br>
+     * <br>
+     * 
+     * Calls for converting non-optional values ({@link Config#getValue} and "Injecting Native Values")
+     * should throw an {@link Exception} for each of the following:<br>
+     * 
+     * 1. {@link IllegalArgumentException} - if the property cannot be converted by the {@link Converter} to the specified type
+     * <br>
+     * 2. {@link NoSuchElementException} - if the property is not defined <br>
+     * 3. {@link NoSuchElementException} - if the property is defined as an empty string <br>
+     * 4. {@link NoSuchElementException} - if the {@link Converter} returns {@code null} <br>
+     * <br>
+     * 
+     * Calls for converting optional values ({@link Config#getOptionalValue} and "Injecting Optional Values")
+     * should only throw an {@link Exception} for #1 ({@link IllegalArgumentException} when the property cannot be converted to
+     * the specified type).
+     */
+    public static <T> T convertValue(String name, String value, Converter<T> converter) {
```

Comment:
Could this code live in `smallrye-config-common`?

Reply:
Why do we need to slip the method?

Reply:
So that it can be called from `smallrye-config-inject` to reduce duplicate code: https://github.com/smallrye/smallrye-config/blob/72a1c0bb48a683c2e50f43743b3f23cf98fd3db4/cdi/src/main/java/io/smallrye/config/inject/ConfigProducerUtil.java#L52